include/OnnxMlirCompiler.h

```diff
@@ -33,47 +39,104 @@
+ *  omSetCompilerOptionsFromArgs, or omSetCompilerOptionsFromEnvAndArgs
+ *  is allowed.
+ *  @param argc Number of input parameters in argv.
+ *  @param argv Array of strings, some of which may be compiler options.
+ *  First argv is ignored as it contains the name of the program.
+ *  @return 0 on success or non-zero error code on failure.
+ */
+ONNX_MLIR_EXPORT int64_t omSetCompilerOptionsFromArgs(
+    int64_t argc, char *argv[]);
+
+/*!
+ *  Define ONNX-MLIR compiler options with options defined by
+ *  the envVarName (default ONNX_MLIR_FLAGS) environment variable
+ *  and the argc/argv parameters. Call expects argv[0] to contain the program
+ *  name. Values not recognized as compiler options result in an error.
+ *  Only a single call to omSetCompilerOptionsFromEnv,
+ *  omSetCompilerOptionsFromArgs, or omSetCompilerOptionsFromEnvAndArgs
+ *  is allowed.
+ *  @param argc Number of input parameters in argv.
+ *  @param argv Array of strings, some of which may be compiler options.
+ *  First argv is ignored as it contains the name of the program.
+ *  @param envVarName Environment variable name, use default when null.
+ *  @return 0 on success or non-zero error code on failure.
+ */
+ONNX_MLIR_EXPORT int64_t omSetCompilerOptionsFromArgsAndEnv(
+    int64_t argc, char *argv[], const char *envVarName);
+
+/*!
+ *  Overwrite the compiler option defined by input parameters.
+ *  Set default value by calling this function before a call to
+ *  omSetCompilerOptionsFromEnv, omSetCompilerOptionsFromArgs, or
+ *  omSetCompilerOptionsFromEnvAndArgs. Or overwrite the current value
+ *  by calling this function after one of the above 3 setter functions.
+ *  @param kind Describe which option kind is being set.
+ *  @param val Value of the option being set. Null pointer reset the
+ *  option.
+ *  @return 0 on success or non-zero error code on failure.
+ */
+ONNX_MLIR_EXPORT int64_t omSetCompilerOption(
+    const OptionKind kind, const char *val);
+
+/*!
+ *  Get the compiler options.
+ *  @param kind Describe which option kind is being set.
+ *  @return Value of compiler option.
+ */
+ONNX_MLIR_EXPORT const char *omGetCompilerOption(
+    const OptionKind kind);
 
 /*!
  *  Compile an onnx model from a file containing MLIR or ONNX protobuf.
  *  @param inputFilename File name pointing onnx model protobuf or MLIR.
  *  @param outputBaseName File name without extension to write output.
  *  @param emissionTarget Target format to compile to.
- *  @param optionKey List of keys specified for the compiler options.
- *  @param optionVal List of string values for corresponding keys.
- *  @param optionNum Number of keys and strings.
+ *  @param options List of compiler options.
  *  @param errorMessage Error message.
  *  @return 0 on success or non-zero error code on failure.
  */
-ONNX_MLIR_EXPORT int omCompileFromFile(const char *inputFilename,
+ONNX_MLIR_EXPORT int64_t omCompileFromFile(const char *inputFilename,
```

Comment:
Why make these int64?

Reply:
For returns it does not matter as much, but using under-defined types (like int) can generate issues, especially when cross compiling with different endianness machines. We just decided for simplicity to use 64-bit long ints for pretty much everything in the interface.